plugins/InspectionGadgets/InspectionGadgetsAnalysis/src/com/siyeh/ig/naming/OverloadedVarargsMethodInspection.java

```diff
@@ -65,10 +75,55 @@ public void visitMethod(@NotNull PsiMethod method) {
                                                         : PsiSubstitutor.EMPTY;
         if (!MethodSignatureUtil.areSignaturesEqual(sameNameMethod.getSignature(substitutor),
                                                     method.getSignature(PsiSubstitutor.EMPTY))) {
+          if (ignoreInconvertibleTypes && !areConvertibleTypesWithVarArgs(method.getParameterList(),
+                                                                          sameNameMethod.getParameterList())) {
+            continue;
+          }
           registerMethodError(method, method);
           return;
         }
       }
     }
+
+    private static boolean areConvertibleTypesWithVarArgs(@NotNull PsiParameterList parameterListWithVarArgs,
+                                                          @NotNull PsiParameterList otherParameterList) {
+      PsiParameter[] parametersWithVarArgs = parameterListWithVarArgs.getParameters();
+      PsiParameter[] otherParameters = otherParameterList.getParameters();
+
+      int lengthForVarArgs = parametersWithVarArgs.length;
+      int otherLength = otherParameters.length;
+
+      //example:
+      //parameterListWithVarArgs: (Integer i1, Integer i2, String... strings)
+      //otherParameterList: (Integer i1)
+      if (lengthForVarArgs > otherLength + 1) {
+        return false;
+      }
+
+      for (int i = 0; i < otherLength; i++) {
+        PsiType type;
+        if (i < lengthForVarArgs) {
+          type = getTypeForComparison(parametersWithVarArgs, i);
+        }
+        else {
+          type = getTypeForComparison(parametersWithVarArgs, lengthForVarArgs - 1);
+        }
+        PsiType otherType = getTypeForComparison(otherParameters, i);
+
+        if (!type.isConvertibleFrom(otherType) && !otherType.isConvertibleFrom(type)) {
+          return false;
+        }
+      }
+
+      return true;
+    }
+  }
+
+  private static PsiType getTypeForComparison(PsiParameter[] parameters, int i) {
+    PsiType type = parameters[i].getType();
+    if (type instanceof PsiEllipsisType) {
+      type = ((PsiEllipsisType)type).getComponentType();
```

Comment:
Perhaps just directly return instead of assigning to the variable. After that the `if` statement can be converted to a conditional expression as well.

Reply:
Thank you! Changed to a conditional expression